File: ror/app/models/flavor.rb
# a Flavor is a property for the vms sizing (tiny: 1 vcpus / 512Mo ram, large: 2 vcpus / 4Go ram, ...)
#
# @author Eric Fehr (eric.fehr@publicis-modem.fr, github: ricofehr)
class Flavor < ActiveRecord::Base
  # multiple different flavors are possible for a project
  has_many :project_flavor, dependent: :destroy
  has_many :projects, through: :project_flavor

  # each vm has one flavvor
  has_many :vms, dependent: :destroy

  # name is mandatory
  validates :title, presence: true
end

File: ror/public/javascripts/models/flavor.js
// Ember model for flavor object
var Flavor = DS.Model.extend({
  title: DS.attr('string'),
  description: DS.attr('string'),
  projects: DS.hasMany('project', {async: true}),
  vms: DS.hasMany('vm', {async: true})
});

module.exports = Flavor;

